samples/dashboards/data/mqtt-connector/demo.js

```diff
@@ -0,0 +1,645 @@
+                highlight: true
+            },
+            dataGridOptions: dataGridOptions
+        }, {
+            renderTo: 'column-chart-2',
+            type: 'Highcharts',
+            connector: {
+                id: 'mqtt-data-2',
+                columnAssignment: [{
+                    seriesId: 'power',
+                    data: ['time', 'power']
+                }]
+            },
+            sync: {
+                highlight: true
+            },
+            chartOptions: {
+                title: {
+                    text: 'Power Station 2'
+                }
+            }
+        }, {
+            renderTo: 'data-grid-2',
+            type: 'DataGrid',
+            connector: {
+                id: 'mqtt-data-2'
+            },
+            sync: {
+                highlight: true
+            },
+            dataGridOptions: dataGridOptions
+        }, {
+            renderTo: 'html-log',
+            type: 'HTML',
+            title: 'MQTT Event Log',
+            html: '<pre id="log-content"></pre>'
+        }, {
+            renderTo: 'app-control',
+            type: 'HTML',
+            html: '<button id="clear-log">Clear log</button>' +
+                '<span id="connect-status">disconnected</span>'
+        }],
+        gui: {
+            layouts: [{
+                rows: [{
+                    // For power station 1
+                    cells: [{
+                        id: 'column-chart-1'
+                    }, {
+                        id: 'data-grid-1'
+                    }]
+                }, {
+                    // For power station 2
+                    cells: [{
+                        id: 'column-chart-2'
+                    }, {
+                        id: 'data-grid-2'
+                    }]
+                }, {
+                    // Log/info area
+                    cells: [{
+                        id: 'html-log'
+                    }]
+                }, {
+                    // Application control
+                    cells: [{
+                        id: 'app-control'
+                    }]
+                }]
+            }]
+        }
+    }, true);
+}
+
+
+/* *
+ *
+ *  Event log and application control functions
+ *
+ * */
+let logContent;
+let connectStatus;
+
+window.onload = () => {
+    // Initialize log
+    logContent = document.getElementById('log-content');
+    logClear();
+    logAppend('Application started');
+
+    // Initialize connection status
+    connectStatus = document.getElementById('connect-status');
+
+    // Add event listener to clear log button
+    const clearButton = document.getElementById('clear-log');
+    clearButton.addEventListener('click', () => {
+        logClear();
+        logAppend('Log cleared by user');
+    });
+};
+
+function logAppend(message) {
+    // Prepend message with timestamp
+    const time = new Date().toLocaleTimeString();
+    logContent.innerHTML = `${time}: ${message}\n` + logContent.innerHTML;
```

Comment:
`.innerHTML` is not recommended for security reasons, and it's slow. Suggest using `.appendChild` or equivalent instead.

Reply:
`innerText` should be enough in this case I guess

Reply:
Ah, didn't notice that it was just plain text in there. Yes, `.innerText `will suffice in this case!